### ISSUE TITLE
Does not compile

### ISSUE THREAD

**Issue by Mc-Pain** [pioneerspacesim/pioneer#5615]
3371981193b5dec438a1db34e400fa8fe62b3bdb does not compile with the following diagnostics:
```
build$ make VERBOSE=1
[...]
cd /home/mcpain/pioneer/build/src/core && /usr/bin/c++ -DFMT_LOCALE -DGLEW_NO_GLU -DGLEW_STATIC -DIMGUI_DEFINE_MATH_OPERATORS -I/home/mcpain/pioneer/build/src/core -I/home/mcpain/pioneer/src/core -I/home/mcpain/pioneer/src -I/home/mcpain/pioneer/contrib -I/home/mcpain/pioneer/contrib/doctest -I/home/mcpain/pioneer/contrib/fmt/include -I/usr/include/freetype2 -I/usr/include/SDL2 -I/usr/include/libpng16 -I/usr/include/sigc++-2.0 -I/usr/lib64/sigc++-2.0/include -I/home/mcpain/pioneer/contrib/glew -I/home/mcpain/pioneer/contrib/lua -I/home/mcpain/pioneer/build -fpermissive -O2 -g -DNDEBUG -std=gnu++17 -fdiagnostics-color=auto -Wall -Wextra -Wno-unused-parameter -Wno-unused-but-set-parameter -Wno-implicit-fallthrough -MD -MT src/core/CMakeFiles/pioneer-core.dir/StringUtils.cpp.o -MF CMakeFiles/pioneer-core.dir/StringUtils.cpp.o.d -o CMakeFiles/pioneer-core.dir/StringUtils.cpp.o -c /home/mcpain/pioneer/src/core/StringUtils.cpp
In file included from /home/mcpain/pioneer/src/core/StringUtils.cpp:8:
/home/mcpain/pioneer/contrib/fmt/include/fmt/core.h:443:7: warning: identifier ‘char8_t’ is a keyword in C++20 [-Wc++20-compat]
  443 | using char8_t FMT_DEPRECATED_ALIAS = internal::char8_type;
      |       ^~~~~~~
In file included from /home/mcpain/pioneer/src/core/StringUtils.cpp:4:
/home/mcpain/pioneer/src/core/StringUtils.h:179:1: error: ‘uint64_t’ does not name a type
  179 | uint64_t StrToUInt64(const std::string &str);
      | ^~~~~~~~
/home/mcpain/pioneer/src/core/StringUtils.h:10:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
    9 | #include <vector>
  +++ |+#include <cstdint>
   10 | 
```

**Comment by Mc-Pain** [comment]
Another one:
```
/home/mcpain/pioneer/src/lua/LuaPushPull.h:17:47: error: ‘uint32_t’ has not been declared
   17 | inline void pi_lua_generic_push(lua_State *l, uint32_t value) { lua_pushinteger(l, value); }
      |                                               
```

**Comment by Web-eWorks** [comment]
I love it when different versions of standard headers implicitly provide a different compilation environment on different systems. Makes it really fun to determine if your tools are lying to you about whether you've included the right headers.

...I'll get these fixed sometime soon, or you could open a quick PR adding `#include <cstdint>` at the bottom of the include blocks of the affected files.

**Comment by Mc-Pain** [comment]
pushed `fix-build` branch: a51976381

**Comment by Web-eWorks** [comment]
Please open that branch as a new PR to this repo and I'll merge it ASAP

**Comment by Web-eWorks** [comment]
Fixed in #5616.
